compiler/test/fail_compilation/fail235.d

```diff
@@ -1,7 +1,7 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/fail235.d(12): Error: expression `typeid(char)` is not a valid template value argument
+fail_compilation/fail235.d(12): Error: template instance `Tuple!(typeid(char))` expression `typeid(char)` is not a valid template value argument
```

Comment:
Why these test changes?

Reply:
Variations in the way the different error functions worked. The current system of a patchwork of slightly different functions is slowly being replaced with one interface, ErrorSink. This PR is progress in that direction.